plan/cacheable_checker.go

```diff
@@ -37,42 +40,15 @@ type cacheableChecker struct {
 
 // Enter implements Visitor interface.
 func (checker *cacheableChecker) Enter(in ast.Node) (out ast.Node, skipChildren bool) {
-	switch node := in.(type) {
-	case *ast.SelectStmt:
-		for _, field := range node.Fields.Fields {
-			if field.Expr == nil {
-				continue
-			}
-			if !checker.canExprBeCached(field.Expr) {
-				checker.cacheable = false
-				return in, true
-			}
-		}
-		if !checker.canExprBeCached(node.Where) {
-			checker.cacheable = false
-			return in, true
-		}
-	case *ast.ExistsSubqueryExpr:
+	switch in.(type) {
+	case *ast.VariableExpr, *ast.ExistsSubqueryExpr:
```

Comment:
Query contains non-deterministic function like `Now()`, `RAND()` should not be cacheable.